File: Graph/dfs.py
n,m = map(int, input(). split())

vis = [False] * (n+1)
def dfs(src, n):
    vis[src] = True
    print(src, end = " ")
    
    for neighbour in graph[src]:
        if not vis[neighbour]:
            dfs(neighbour, n)
        

graph = [[]for _ in range(n+1)]
for _ in range(m):
    u,v = map(int, input(). split())
    graph[u].append(v)
    graph[v].append(u)
    
dfs(1,n)

File: Graph/dfs.cpp
#include <bits/stdc++.h>
using namespace std;

typedef vector<int> vi;
typedef vector<bool> vb;
#define MAX 100

vi graph[MAX];
vb vis(MAX, false);

void addEdge(int u, int v, bool dir = false)
{
    graph[u].push_back(v);
    if (!dir)
    {
        graph[v].push_back(u);
    }
}

void dfs(int src)
{
    vis[src] = true;
    cout << src << " ";
    for (int neighbour : graph[src])
    {
        if (!vis[neighbour])
        {
            dfs(neighbour);
        }
    }
}

int main(void)
{
    int n, m;
    cin >> n >> m;
    for (int i = 0; i < m; i++)
    {
        int u, v;
        cin >> u >> v;
        addEdge(u, v);
    }
    dfs(1);
    return 0;
}